File: tests/fixtures/generic/vulnerable/taint_path_traversal.java
// @audit-fixture
// @rule: taint_path_traversal
// @expected: detected
import javax.servlet.http.*;
import java.nio.file.*;
public class FileServlet extends HttpServlet {
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws Exception {
        String file = request.getParameter("file");
        byte[] data = Files.readAllBytes(Paths.get("/data/" + file));
        response.getOutputStream().write(data);
    }
}

File: tests/fixtures/generic/vulnerable/taint_path_traversal.cs
// @audit-fixture
// @rule: taint_path_traversal
// @expected: detected
public class FileController : Controller {
    public IActionResult Download() {
        var file = Request.Query["file"];
        var content = System.IO.File.ReadAllText(file);
        return Content(content);
    }
}
